controllers/array_action/array_mediator_svc.py

```diff
@@ -591,20 +592,29 @@ def get_snapshot(self, volume_id, snapshot_name, pool, is_virt_snap_func):
             return None
         return self._generate_snapshot_response_with_verification(target_cli_volume)
 
-    def get_object_by_id(self, object_id, object_type):
-        cli_object = self._get_cli_volume_by_wwn(object_id)
-        if not cli_object:
+    def get_object_by_id(self, object_id, object_type, is_virt_snap_func=False):
+        if is_virt_snap_func and object_type == controller_config.SNAPSHOT_TYPE_NAME:
+            cli_snapshot = self._get_cli_snapshot_by_id(object_id)
+            if not cli_snapshot:
+                return None
+            source_cli_volume = self._get_cli_volume(cli_snapshot.volume_name)
+            if not source_cli_volume:
+                return None
+            return self._generate_snapshot_response_from_cli_snapshot(cli_snapshot, source_cli_volume)
+        cli_volume = self._get_cli_volume_by_wwn(object_id)
+        if not cli_volume:
             return None
         if object_type is controller_config.SNAPSHOT_TYPE_NAME:
-            return self._generate_snapshot_response_with_verification(cli_object)
-        cli_volume = self._get_cli_volume(cli_object.name)
+            return self._generate_snapshot_response_with_verification(cli_volume)
+        cli_volume = self._get_cli_volume(cli_volume.name)
```

Comment:
this is so weird - do we have a ticket for `filtered_cli_volume` renaming?

Reply:
not that I know

Reply:
could you open one and write its number here?

Reply:
CSI-4507